Ixfleura/Commands/Modules/SuggestionModule.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Threading.Tasks;
+using Disqord;
+using Disqord.Bot;
+using Disqord.Rest;
+using Ixfleura.Commands.Checks;
+using Ixfleura.Common.Extensions;
+using Ixfleura.Data.Entities;
+using Ixfleura.Services;
+using Microsoft.Extensions.Configuration;
+using Qmmands;
+
+namespace Ixfleura.Commands.Modules
+{
+    [Group("suggest", "suggestion")]
+    public class SuggestionModule : DiscordGuildModuleBase
+    {
+        private readonly IConfiguration _configuration;
+        private readonly SuggestionService _suggestionService;
+
+        public SuggestionModule(IConfiguration configuration ,SuggestionService suggestionService)
+        {
+            _configuration = configuration;
+            _suggestionService = suggestionService;
+        }
+
+        [Command]
+        public async Task CreateSuggestionAsync([Remainder] string content)
+        {
+            var suggestion = new Suggestion
+            {
+                Content = content,
+                GuildId = Context.GuildId,
+                SuggesterId = Context.Author.Id
+            };
+
+            var suggestionChannelId = _configuration.GetValue<ulong>("fundamics:suggestion_id");
+
+            await _suggestionService.CreateSuggestionAsync(suggestion);
+
+            var suggestionEmbed = new LocalEmbed()
+                .WithThumbnailUrl(Context.Author.GetAvatarUrl(ImageFormat.Png))
+                .WithFooter($"suggestion Id: {suggestion.Id}")
+                .WithIxColor()
+                .AddField("Submitter", Context.Author.Tag)
+                .AddField("Suggestion", content);
+
+            // for testing just change this to Context.ChannelId
+            var suggestionMessage = await Context.Bot.SendMessageAsync(suggestionChannelId, new LocalMessage().WithEmbed(suggestionEmbed));
+
+            var (denyEmoji, checkEmoji) = _suggestionService.GetSuggestionEmojis();
+
+            await suggestionMessage.AddReactionAsync(checkEmoji);
+            await suggestionMessage.AddReactionAsync(denyEmoji);
+            
+            suggestion.MessageId = suggestionMessage.Id;
+            suggestion.ChannelId = suggestionMessage.ChannelId;
+            
+            await _suggestionService.UpdateSuggestionAsync(suggestion);
+
+            await Context.Message.DeleteAsync();
+        }
+
+        [Command("accept")]
+        [RequireModOrAdmin]
+        public async Task AcceptSuggestionAsync(int id, [Remainder] string reason)
+        {
+            var suggestion = await _suggestionService.GetSuggestionAsync(id);
+
+            if (suggestion is null)
+            {
+                var message = await Response("A suggestion with that id could not be found.");
+                _ = Task.Delay(8000).ContinueWith(_ => message.DeleteAsync());
```

Comment:
It would make more sense to just await the delay here because commands aren't blocking in Disqord
```suggestion
               await Task.Delay(8000);
               await message.DeleteAsync()
```